test/models.py

```diff
@@ -67,3 +67,28 @@ def predict_proba(self, X_pred):
         prediction = self.predict(X_pred).reshape(-1, 1)
         zeros = np.repeat(0, X_pred.shape[0]).reshape(-1, 1)
         return np.concatenate((1 - prediction, prediction, zeros), axis=1)
+
+
+class PredictAsDataFrameTestModel(object):
+    def __init__(self, return_predictions_as_dataframe=True):
+        self.return_predictions_as_dataframe = return_predictions_as_dataframe
+        pass
+
+    def fit(self, X, y):
+        pass
+
+    def predict(self, X_pred):
+        result = np.repeat(0, X_pred.shape[0])
+        result[0] = 1
+        if self.return_predictions_as_dataframe:
+            return pd.DataFrame(result)
```

Comment:
crazy to see this output a dataframe as the prediction

Reply:
I know. But it is good to cover this case.